tianshou/data/batch.py

```diff
@@ -520,9 +521,9 @@ def stack_(self,
         values_shared = [
             [e[k] for e in batches] for k in keys_shared]
         for k, v in zip(keys_shared, values_shared):
-            if isinstance(v[0], (dict, Batch)):
+            if all(isinstance(e, (dict, Batch)) for e in v):
```

Comment:
Same.

Reply:
I profile the code and find that this pr will not affect `stack_` and `to_torch`'s behavior in the test script.